### ISSUE TITLE
[issue] bugs in race condition module

### ISSUE THREAD

**Issue by ehsandeep** [projectdiscovery/nuclei#576]
**Describe the bug**
With the following changes https://github.com/projectdiscovery/nuclei/issues/483, I have identified new and similar issue:-

- [x] Request never gets completed
- [x] Initial single post request and remaining later, the expected behavior is that all requests will be posted at the end. 

```
echo http://0.0.0.0:8000 | ./nuclei -t race.yaml
```

**Nuclei version**
 dev

<details>
<summary>Test Template</summary>

```yaml
id: race-condition-testing

info:
  name: Race condition testing
  author: pdteam
  severity: info

requests:
  - raw:
      - |
        POST /coupons HTTP/1.1
        Host: {{Hostname}}
        Pragma: no-cache
        Cache-Control: no-cache, no-transform
        User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0

        promo_code=20OFF        

    race: true
    race_count: 10

    matchers:
      - type: status
        part: header
        status:
          - 200
```
</details>

**Comment by Mzack9999** [comment]
This should be fixed in dev with #575

**Comment by ehsandeep** [comment]
This is fixed with https://github.com/projectdiscovery/nuclei/pull/590
